src/sif/truckcost.cc

```diff
@@ -310,6 +336,23 @@ TruckCost::TruckCost(const Costing& costing)
     speedfactor_[s] = (kSecPerHour * 0.001f) / static_cast<float>(s);
   }
 
+  // Sets the surface factor for truck
+  surface_factor_ = 0.5f;
```

Comment:
this doesn't make much sense here. also shouldn't truck be a little more sensitive to surfaces? IMO this can easily be `1.0f` in which case you can omit the whole `surface_factor_` and only use the array for costing further down

Reply:
I did that because the surface factor was `1` only for Motorcycle in `Autocost.cc`.

https://github.com/valhalla/valhalla/blob/6bdc1e64eaa2196697b5d07b7735cc0dc465adaa/src/sif/autocost.cc#L355-L371

Reply:
I did that because the surface factor was `1` only for Motorcycle in `Autocost.cc`.

https://github.com/valhalla/valhalla/blob/6bdc1e64eaa2196697b5d07b7735cc0dc465adaa/src/sif/autocost.cc#L355-L371

Reply:
you just uncovered another undocumented parameter:) `type` for car, where `motorcycle` seems odd and probably from days before there was a proper costing model for it. 

anyways, I'm also not sure if 0.5 or 1.0 is better here, though leaning towards 1.0. my point is we don't need that variable because a truck is a truck (for us so far at least), so you can just take that factor directly into the `EdgeCost` and remove the definition of `surface_factor_`